src/app/shared/formly/types/date-picker-field/date-picker-field.component.ts

```diff
@@ -0,0 +1,37 @@
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { NgbDate, NgbDateStruct } from '@ng-bootstrap/ng-bootstrap';
+import { FieldType, FieldTypeConfig } from '@ngx-formly/core';
+import { Observable, isObservable } from 'rxjs';
+
+import { DateHelper } from 'ish-core/utils/date-helper';
+
+/**
+ * Type for a  date-picker field.
+ *
+ * @defaultWrappers 'form-field-horizontal', 'validation'
+ *
+ */
+@Component({
+  selector: 'ish-date-picker-field',
+  templateUrl: './date-picker-field.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DatePickerFieldComponent extends FieldType<FieldTypeConfig> {
+  readonly isObservableFn = isObservable;
+
+  get minDate$(): Observable<NgbDateStruct> {
+    return DateHelper.addDaysToTodayAsync(this.to.minDays);
+  }
+
+  get maxDate$(): Observable<NgbDateStruct> {
+    return DateHelper.addDaysToTodayAsync(this.to.maxDays);
+  }
+
+  markDateDisable(isSatExcluded: unknown, isSunExcluded: unknown) {
```

Comment:
because then tslint would complain about this ```const disableSaturday = isSatExcluded === true ? DateHelper.isSaturday(date) : false;``` => unnecessarily compares a boolean

however this statement  ```isSatExcluded === true```  is a guard that needs to be there  to sanitize the input.

and having an expression like ```!!isSatExcluded ``` would fail as a guard because ```!!(valid number)``` will also be true.